custom_components/tuya_local/devices/poolex_arctic_action.yaml

```diff
@@ -0,0 +1,268 @@
+name: Pool heatpump
+products:
+  - id: ea8w1w
+    name: Poolex Arctic Action
+primary_entity:
+  entity: climate
+  icon: "mdi:hot-tub"
+  translation_key: pool_heatpump
+  dps:
+    - id: 1
+      type: boolean
+      name: hvac_mode
+      mapping:
+        - dps_val: false
+          value: "off"
+        - dps_val: true
+          constraint: preset_mode
+          conditions:
+            - dps_val: Heat_Intelligent
+              value: heat
+            - dps_val: Heat_Powerful
+              value: h_powerful
+            - dps_val: Heat_Energy_Saving
+              value: h_silent
+            - dps_val: Cool_Intelligent
+              value: cool
+            - dps_val: Cool_Powerful
+              value: c_powerful
+            - dps_val: Cool_Silent
+              value: c_silent
+    - id: 2
+      type: integer
+      name: temperature
+      range:
+        min: 5
+        max: 40
+      mapping:
+        - constraint: temperature_unit
+          conditions:
+            - dps_val: f
+              range:
+                min: 41
+                max: 104
+              value_redirect: temp_set_f
+    - id: 3
+      type: integer
+      name: current_temperature
+      mapping:
+        - constraint: temperature_unit
+          conditions:
+            - dps_val: f
+              value_redirect: temp_current_f
+    - id: 4
+      type: string
+      name: preset_mode
+      mapping:
+        - dps_val: Heat_Intelligent
+          icon: "mdi:hot-tub"
+          value: heat
```

Comment:
preset_mode can be anything, but it is recommended to take from existing translations, in this cas the translation_key pool_heatpump was already defined, and the presets defined for other pool heatpumps at https://github.com/make-all/tuya-local/blob/main/custom_components/tuya_local/translations/en.json#L128 closely match (smart_heat for Heat_Intelligent, quick_heat for Heat_Powerful, and quiet_heat for Heat_Energy_Saving, and likewise for cool.  The Energy_Saving does not quite match quiet, but it seems like there is a common controller, and manufacturers choose the naming based on how they want to market it).